ql/src/java/org/apache/hadoop/hive/ql/optimizer/GenMapRedUtils.java

```diff
@@ -1368,7 +1368,7 @@ public static void createMRWorkForMergingFiles(FileSinkOperator fsInput,
     // NOTE: we should gather stats in MR1 rather than MR2 at merge job since we don't
     // know if merge MR2 will be triggered at execution time
     MoveWork dummyMv = null;
-    if (srcMmWriteId == null) {
+    if (srcMmWriteId == null && !fsInputDesc.isIcebergTable()) {
```

Comment:
@SourabhBadhya does it look valid to you? 
What I noticed is that we create some dummy moveWork that tries to move an empty `.hive-staging../-tmp.-ext` that doesn't exist. 
In case of `UNION ALL` we create UNION subdirs under  `.hive-staging../-ext` that cause the root table folder wipe.

Reply:
Yeah, moveTask for Iceberg is handled in a different way. Its ok for now, however ideally this should be handled on the move side. Was there any mergeWork created here?

Reply:
yes, dummy LoadFileWork

Reply:
Honestly, I am fine with this. It's an overkill to define a function from Storage handler or custom interface just to see if it's an Iceberg table.

Reply:
`fsInputDesc` has similar methods for ACID and insert-only tables